pkg/ocm/upgrades.go

```diff
@@ -96,3 +96,51 @@ func (c *Client) CancelUpgrade(clusterID string) (bool, error) {
 	}
 	return true, nil
 }
+
+func (c *Client) GetMissingGateAgreements(
+	clusterID string,
+	upgradePolicy *cmv1.UpgradePolicy) ([]*cmv1.VersionGate, error) {
+	response, err := c.ocm.ClustersMgmt().V1().Clusters().
+		Cluster(clusterID).UpgradePolicies().Add().Parameter("dryRun", true).Body(upgradePolicy).Send()
+
+	if err != nil {
+		if response.Error() != nil {
+			// parse gates list
+			errorDetails, ok := response.Error().GetDetails()
+			if !ok {
+				return []*cmv1.VersionGate{}, handleErr(response.Error(), err)
+			}
+			data, err := json.Marshal(errorDetails)
+			if err != nil {
+				return []*cmv1.VersionGate{}, handleErr(response.Error(), err)
+			}
+			gates, err := cmv1.UnmarshalVersionGateList(data)
```

Comment:
I wonder how this works. 

The returned value is not in the form of API list, with `Items` etc.